pkg/resource/labels.go

```diff
@@ -27,28 +32,130 @@ func (labels Labels) Equal(other Labels) bool {
 
 // Matches if labels match the LabelTerm.
 func (labels Labels) Matches(term LabelTerm) bool {
-	if labels.KV.Empty() {
-		return term.Op == LabelOpNotExists
+	matches := labels.matches(term)
+
+	if term.Invert {
+		return !matches
 	}
 
-	switch term.Op {
-	case LabelOpNotExists:
-		_, ok := labels.Get(term.Key)
+	return matches
+}
 
-		return !ok
-	case LabelOpExists:
-		_, ok := labels.Get(term.Key)
+func (labels Labels) matches(term LabelTerm) bool {
+	if labels.KV.Empty() && term.Op == LabelOpExists {
+		return false
+	}
+
+	value, ok := labels.Get(term.Key)
 
-		return ok
+	if !ok {
+		return false
+	}
+
+	switch term.Op {
+	case LabelOpExists:
+		return true
 	case LabelOpEqual:
-		value, ok := labels.Get(term.Key)
+		return value == term.Value
+	case LabelOpIn:
+		return slices.Contains(term.Set, value)
+	case LabelOpLTE:
+		return value <= term.Value
+	case LabelOpLT:
+		return value < term.Value
+	case LabelOpLTNumeric:
+		left, right, ok := getNumbers(value, term.Value)
+		if !ok {
+			return false
+		}
 
+		return left < right
+	case LabelOpLTENumeric:
+		left, right, ok := getNumbers(value, term.Value)
 		if !ok {
 			return false
 		}
 
-		return value == term.Value
+		return left <= right
 	default:
 		panic(fmt.Sprintf("unsupported label term operator: %v", term.Op))
 	}
 }
+
+var (
+	re    = regexp.MustCompile(`^(\d+)(.*)$`)
```

Comment:
let's not use global variables as `regexp`, this causes allocations on import even if you never use it. and I think this can be easily parsed with a straightfowards Go function which will runs faster